src/features/filter.ts

```diff
@@ -1,10 +1,38 @@
-export const actions = { /* put action creators here */};
+import { Status } from '../types/Status';
 
-const filterReducer = () => {
-  return {
-    query: '',
-    status: 'all',
-  };
+type Select = { type: 'filter/SELECT', payload: Status };
+type Query = { type: 'filter/QUERY', payload: string };
+
+const setSelect = (value: Status): Select => (
+  { type: 'filter/SELECT', payload: value }
+);
+const setQuery = (value: string): Query => (
+  { type: 'filter/QUERY', payload: value }
+);
+
+type Actions = Select | Query;
+
+export type Filter = {
+  query: string,
+  status: Status,
+};
+
+export const actions = { setSelect, setQuery };
+
+const filterReducer = (filter: Filter, action: Actions) => {
+  switch (action.type) {
+    case 'filter/SELECT':
+      return { ...filter, status: action.payload };
+
+    case 'filter/QUERY':
+      return { ...filter, query: action.payload };
+
+    default:
+      return filter || {
+        query: '',
+        status: 'active',
+      };
```

Comment:
you can just specify the initial state for this reducer

Reply:
@etojeDenys Hi) When modal window opened list and filter params reset(